cardinal/message_manager.go

```diff
@@ -9,22 +9,30 @@ import (
 	"pkg.world.dev/world-engine/cardinal/types"
 )
 
-type Manager struct {
+type MessageManager interface {
+	RegisterMessage(msgType types.Message, msgReflectType reflect.Type) error
+	GetRegisteredMessages() []types.Message
+	GetMessageByID(id types.MessageID) types.Message
+	GetMessageByFullName(fullName string) (types.Message, bool)
+	GetMessageByType(mType reflect.Type) (types.Message, bool)
+}
+
+type MessageManagerImpl struct {
```

Comment:
this struct doesn't need to be public. just follow the naming convention from the system.go (https://github.com/Argus-Labs/world-engine/blob/main/cardinal/system.go)